### DISCUSSION TITLE
Passing variables into function used in request.dynamic ?

### DISCUSSION THREAD

**Discussion by Russsgithub** [savonet/liquidsoap#4026] [General]
Does anyone know of a way of passing vars into the function used in request dynamic.

I have a few scripts that use pretty much the same function to call new tracks from an api call, where the only difference is the api end point.

It would be cleaner for me to able to use the same function and pass in variable to define the endpoint to be called . At pressent I can't see a way of doing this.

for example

```
def get_next_track(stream)
  api_uri = base_uri^"/"^stream
  track_uri = http.get(api_uri)
  ....
  request.create(track_uri)
end

music = request.dynamic(get_next_track("music"))
extras = request.dynamic(get_next_track("extras"))
```
This will fail as request.dynamic requires a function without the parentheses, and means i have to have two function , one for each request.dynamic operator.

Thanks

**Comment by vitoyucepi** [comment]
Hi @Russsgithub,
The `request.dynamic` function expects an argument of type function, with no arguments, that returns `request`. You could create an anonymous function to do this, or return a function from the `get_next_track`.

Here's a variant using an anonymous function
```ruby
music = request.dynamic({get_next_track("music")})
extras = request.dynamic({get_next_track("extras")})
```

Also be aware that in some cases `request.dynamic` may start requesting the next track every frame. So you should guard the `http.get` call.

**Reply by Russsgithub** [reply]
@vitoyucepi , Ok, that's exciting. I shall give it a go.

Thanks again.

Russ
